cluster/transport/client.go

```diff
@@ -160,6 +155,12 @@ func (cl *Client) getConn(leaderAddress string) (*grpc.ClientConn, error) {
 		return nil, fmt.Errorf("dial: %w", err)
 	}
 
+	if cl.leaderConn != nil {
```

Comment:
Wouldn't make more sense to close existing conn before grpc.Dial() to avoid having two connections open simultaneously

Reply:
I agree it is a bit off, but the old connection should be for different node(leader address), that will happen in case if the leader changed but if we weren't able to connect to the new leader we could end up hanging on the previous leader connections while we failed to connect to the new one, we may won't need to merge it after all, but i am experimenting if we are going to need to

Reply:
Note that the new connection will be available only once the connection to the old leader has been closed.